ReSharper.FSharp/src/FSharp/FSharp.Psi.Services/src/Util/PsiUtil.fs

```diff
@@ -409,6 +409,22 @@ let rec skipIntermediateParentsOfSameType<'T when 'T :> ITreeNode and 'T: not st
 let rec skipIntermediatePatParents (fsPattern: IFSharpPattern) =
     skipIntermediateParentsOfSameType<IFSharpPattern> fsPattern
 
+let rec getQualifiedExprIgnoreParens (expr: IFSharpExpression) =
+    match QualifiedExprNavigator.GetByQualifier(expr.IgnoreParentParens()) with
+    | null -> expr.IgnoreParentParens()
+    | expr -> getQualifiedExprIgnoreParens expr
+
+let rec getQualifiedExpr (expr: IFSharpExpression) =
+    match QualifiedExprNavigator.GetByQualifier(expr) with
+    | null -> expr
+    | expr -> getQualifiedExprIgnoreParens expr
+
+let isDirectPartOfDotLambda (expr: IFSharpExpression) =
+    let prefixApp = PrefixAppExprNavigator.GetByExpression(expr)
```

Comment:
Todo: the argument should not be considered as a direct part, and we need a general fix for introduce refactoring logic here.